Test of these changes:

```
$ ./test/sanity/validate-modules/validate-modules lib/ansible/modules/cnos_factory.py
============================================================================
lib/ansible/modules/cnos_factory.py
============================================================================
TRACE:
    Traceback (most recent call last):
      File "./test/sanity/validate-modules/validate-modules", line 566, in _validate_docs
        get_docstring(self.path, verbose=True)
      File "/Users/matt/python_venvs/ansibledev/ansible/lib/ansible/utils/module_docs.py", line 120, in get_docstring
        raise Exception("Attempt to extend a documentation fragement (%s) of unknown type: %s" % (fragment_name, filename))
    Exception: Attempt to extend a documentation fragement (cnos) of unknown type: lib/ansible/modules/cnos_factory.py
ERROR:304:Unknown DOCUMENTATION error, see TRACE
ERROR:305:DOCUMENTATION.options must be a dictionary/hash when used with DOCUMENTATION.extends_documentation_fragment
```

--- a/test/sanity/validate-modules/validate-modules
+++ b/test/sanity/validate-modules/validate-modules
@@ -577,7 +577,7 @@ class ModuleValidator(Validator):
                             'Unknown DOCUMENTATION error, see TRACE'
                         ))
 
-                if doc.get('options') is None and doc.get('extends_documentation_fragment'):
+                if 'options' in doc and doc['options'] is None and doc.get('extends_documentation_fragment'):
                     self.errors.append((
                         305,
                         ('DOCUMENTATION.options must be a dictionary/hash when used '

